music_assistant/controllers/metadata.py

```diff
@@ -302,24 +303,29 @@ async def get_image_data_for_item(
 
     async def get_image_url_for_item(
         self,
-        media_item: MediaItemType,
+        media_item: MediaItemType | ItemMapping,
         img_type: ImageType = ImageType.THUMB,
         resolve: bool = True,
     ) -> str | None:
         """Get url to image for given media media_item."""
         if not media_item:
             return None
+
         if isinstance(media_item, ItemMapping):
-            media_item = await self.mass.music.get_item_by_uri(media_item.uri)
-        if media_item and media_item.metadata.images:
-            for img in media_item.metadata.images:
-                if img.type != img_type:
-                    continue
-                if img.remotely_accessible and not resolve:
-                    continue
-                if img.remotely_accessible and resolve:
-                    return self.get_image_url(img)
-                return img.path
+            # Check if the ItemMapping already has an image - avoid expensive API call
+            if media_item.image and media_item.image.type == img_type:
+                if media_item.image.remotely_accessible and resolve:
+                    return self.get_image_url(media_item.image)
+                elif not media_item.image.remotely_accessible:
+                    return media_item.image.path
```

Comment:
this should be the other way around
```suggestion
                if not media_item.image.remotely_accessible and resolve:
                    return self.get_image_url(media_item.image)
                else:
                    return media_item.image.path
```